app/controllers/util/FormField.scala

```diff
@@ -255,12 +254,16 @@ object DateField {
   )
 
   def validate(day: String, month: String, year: String, dateField: DateField): Option[String] = {
-
-    var error = validateValue(day, "Day", invalidDayValidation)
-    error = if (error.isEmpty) validateValue(month, "Month", invalidMonthValidation) else error
-    error = if (error.isEmpty) validateValue(year, "Year", invalidYearValidation) else error
-    error = if (error.isEmpty) checkDayForTheMonthAndYear(day.toInt, month.toInt, year.toInt) else error
-    if (error.isEmpty) checkIfFutureDateIsAllowed(day.toInt, month.toInt, year.toInt, dateField) else error
+    val fieldName = if (dateField.fieldAlternativeName.isEmpty) dateField.fieldName.toLowerCase else dateField.fieldAlternativeName
+    if (day.isEmpty && month.isEmpty && year.isEmpty) {
+      Some(dateNotEnteredError.format(fieldName))
+    } else {
+      var error = validateValue(day, "Day", invalidDayValidation, "3, 15, 21", "31", fieldName)
```

Comment:
Can you replace the `var` with a series of vals please. We try not to use `var` except when it's unavoidable and I think you can do this without it. Something like:
```scala
      val dayError = validateValue(day, "Day", invalidDayValidation, "3, 15, 21", "31", fieldName)
      val monthError = if (dayError.isEmpty) validateValue(month, "Month", invalidMonthValidation, "3, 9, 12", "12", fieldName) else dayError
      val yearError = if (monthError.isEmpty) validateValue(year, "Year", invalidYearValidation, "1994, 2000, 2023", "", fieldName) else monthError
      val dayForMonthAndYearError = if (yearError.isEmpty) checkDayForTheMonthAndYear(day.toInt, month.toInt, year.toInt, fieldName) else yearError
      if (dayForMonthAndYearError.isEmpty) checkIfFutureDateIsAllowed(day.toInt, month.toInt, year.toInt, dateField) else dayForMonthAndYearError
```

Reply:
ah my bad, the var was there prior, your above suggestion works fine :+1: